dev/Deployment/DeploymentManager.cpp

```diff
@@ -165,12 +166,12 @@ namespace winrt::Microsoft::Windows::ApplicationModel::WindowsAppRuntime::implem
         winrt::Microsoft::Windows::ApplicationModel::WindowsAppRuntime::DeploymentInitializeOptions const& deploymentInitializeOptions,
         bool isRepair)
     {
-        if (isAutoInitialized)
+        auto lock{ m_isInitializedLock.lock_exclusive() };
```

Comment:
I don't think this is required in practice, given entry point startup is single-threaded.  But a little paranoia never hurts.

Reply:
>given entry point startup is single-threaded

What entry point? Nothing stops 2 threads from instantiating DeploymentManager and calling Initialize concurrently. Hence the standard guidance to protect globals against concurrency misuse.

Unlikely a problem, for well intentioned devs, but these are the sorts of things malware looks for. Best to follow the usual guidance than learn later someone intentionally forced the dice to come up snakeeyes for fun and profit.